problem/src/main/java/org/zalando/problem/Status.java

```diff
@@ -263,11 +266,36 @@
     private final int code;
     private final String reason;
 
+    // Build a HashMap of all status keyed by their codes.
+    // Used in the `ofCode` factory method
+    private static Map<Integer, Status> allStatusByCode = new HashMap<>();
```

Comment:
On a second though, just leave it with the linear search for now. I'll change it in a separate PR. I just realized that we could simplify the jackson module as well: https://github.com/zalando/problem/blob/master/jackson-datatype-problem/src/main/java/org/zalando/problem/StatusTypeDeserializer.java

Reply:
done.